### PR TITLE
Jwt revocation check

### PR DESCRIPTION
**Pull request by yogsototh**
> **Epic** https://github.com/threatgrid/iroh/issues/2897
> Close https://github.com/threatgrid/iroh/issues/2925

Ability to add a check on the JWT via a revocation endpoint (depending on the issuer).

<a name="qa">[§](#qa)</a> QA
============================

1. Revoke a JWT in CTR
2. Check you cannot use this JWT to make call to CTIA

<a name="ops">[§](#ops)</a> Ops
===============================

- TODO Config change needed: threatgrid/tenzin#

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
public: Intel support JWT revocation.
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/entity/web_test.clj [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+(defn jwt-url-checks-test
+  [url-1 url-2 tst-fn]
+  (testing "JWT Key Map + URL-Check"
+    (apply-fixtures
+     ["ctia.http.jwt.enabled" true
+      "ctia.http.jwt.public-key-map"
+      "IROH Auth=resources/cert/ctia-jwt.pub,IROH Auth TEST=resources/cert/ctia-jwt-2.pub"
+
+      "ctia.http.jwt.url-check.endpoints" (str "IROH Auth=" url-1 ",IROH Auth TEST=" url-2)
+      "ctia.http.jwt.url-check.timeout" 1000
```
Comment (msprunck) [inline test/ctia/entity/web_test.clj]: You could use less than `1000` to reduce the `Thread.sleep`

File: test/ctia/entity/web_test.clj [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+                          :external_ids ["http://ex.tld/ctia/judgement/judgement-123"
+                                         "http://ex.tld/ctia/judgement/judgement-456"]
+                          :disposition 2
+                          :source "test"
+                          :priority 100
+                          :timestamp #inst "2042-01-01T00:00:00.000Z"
+                          :severity "High"
+                          :confidence "Low"
+                          :reason "This is a bad IP address that talked to some evil servers"
+                          :valid_time {:start_time "2016-02-11T00:40:48.212-00:00"}}
```
Comment (ereteog) [inline test/ctia/entity/web_test.clj]: `:body new-judgement-1`

File: test/ctia/entity/web_test.clj [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+                                :external_ids ["http://ex.tld/ctia/judgement/judgement-123"
+                                               "http://ex.tld/ctia/judgement/judgement-456"]
+                                :id "http://localhost:3001/ctia/judgement/judgement-00001111-0000-1111-2222-000011112222"
+                                :disposition 2
+                                :source "test"
+                                :priority 100
+                                :severity "High"
+                                :confidence "Low"
+                                :reason "This is a bad IP address that talked to some evil servers"
+                                :valid_time {:start_time "2016-02-11T00:40:48.212-00:00"}}
```
Comment (ereteog) [inline test/ctia/entity/web_test.clj]: `:body new-judgement-1` :-)

Comment (yogsototh) [reply test/ctia/entity/web_test.clj]: not this one, there is an `id` field added.

**Pull request merged**